### ISSUE TITLE
VT100 syntax highlighting within source code files

### ISSUE THREAD

**Issue by dfaust** [TobiasFaller/vscode-vt100-syntax-highlight#30]
Thanks for this great extension!

It works very well. Unfortunately though, I need an extension that works in combination with other source code highlighting extensions on my source code files.
I created a fork and changed it to fit my needs. But maybe you want to enhance your extension to support this use case natively.

Feel free to close this issue if you're not interested in that. I just wanted to say hi.

https://github.com/dfaust/vscode-vt100-syntax-highlight

**Comment by TobiasFaller** [comment]
Hey @dfaust,

can you give me an example of how the combination looks like?

I've also seen you changed those lines:
```typescript
 if (backgroundColor != 'default') {
    appliedDecorations.get('background-color-' + backgroundColor)!.push(range);
}
 if (foregroundColor != 'default') {
    appliedDecorations.get('foreground-color-' + foregroundColor)!.push(range);
}
```
Is that an optimization for speed?

Regards,
Tobias

**Comment by dfaust** [comment]
Here is a screenshot of my patched `vscode-vt100-syntax-highlight` version in combination with [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer).

![image](https://user-images.githubusercontent.com/893710/224542764-f4f1fbcb-541d-4d76-80cf-d2dc9496cd26.png)

I changed the lines you mentioned in order to not override the Rust syntax highlighting. Otherwise everything but the VT100 text is just white.

**Comment by TobiasFaller** [comment]
Hey @dfaust,

I've integrated your feature into the extension.
However, there were some problems that I had to work around:

The highlighting is not based on the underlying grammar of the language.
This means that in most cases the highlighting just works.

![image](https://github.com/TobiasFaller/vscode-vt100-syntax-highlight/assets/1679294/c31b1e79-83c9-48c8-8cfa-0bda00a633f1)

However, when the escape sequence is not terminated then the rest of the documented will be highlighted, independent of the termination of the string / comment. Therefore, I included an include / exclude option to the extension to enable / disable highlighting for specific languages to not interfere with the editor when undesired. The new version will be available soon.

![image](https://github.com/TobiasFaller/vscode-vt100-syntax-highlight/assets/1679294/bcde1246-1946-4634-8bcf-544172776b1e)

Best Regards,
Tobias Faller
